### PR TITLE
Features/upgradable separated

### PR DESCRIPTION
**Pull request by Nakasar**
The main goal of this PR is to keep contracts ERC734 and ERC735 standalone (ie. being able to deploy a standalone ERC734 without any code from the ERC735, not valid the other way around).

To do so, I have made the Struct.sol and Storage.sol only inherited by the Identity contract.

I also removed oz project.json and migrations.sol.

### CODE REVIEW

File: contracts/Identity.sol [hunk 1]
```diff
@@ -1,25 +1,271 @@
...
 
 /**
  * @dev Implementation of the `IERC734` "KeyHolder" and the `IERC735` "ClaimHolder" interfaces into a common Identity Contract.
+ * This implementation has a separate contract were it declares all storage, allowing for it to be used as an upgradable logic contract.
  */
-contract Identity is ERC734, IIdentity, Version {
-
-
-
+contract Identity is Structs, Storage, LibraryLockDataLayout, LibraryLock, IIdentity, Version {
```
Comment (Aboudjem) [inline contracts/Identity.sol:14]: Storage already inherits from Structs

File: contracts/erc735/ERC735.sol [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    *      - issuer: The issuers identity contract address, or the address used to sign the above signature. If an identity contract, it should hold the key with which the above message was signed, if the key is not present anymore, the claim SHOULD be treated as invalid. The issuer can also be a contract address itself, at which the claim can be verified using the call data.
+    *      - signature: Signature which is the proof that the claim issuer issued a claim of topic for this identity. it MUST be a signed message of the following structure: `keccak256(abi.encode(identityHolder_address, topic, data))`
+    *      - data: The hash of the claim data, sitting in another location, a bit-mask, call data, or actual data based on the claim scheme.
+    *      - uri: The location of the claim, this can be HTTP links, swarm hashes, IPFS hashes, and such.
+    */
+    struct Claim {
```
Comment (Aboudjem) [inline contracts/erc735/ERC735.sol:24]: This struct should be included in Storage

File: contracts/erc735/ERC735.sol [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        uint256 topic;
+        uint256 scheme;
+        address issuer;
+        bytes signature;
+        bytes data;
+        string uri;
+    }
+
+    mapping(bytes32 => Claim) internal claims;
```
Comment (Aboudjem) [inline contracts/erc735/ERC735.sol:33]: both mappings should be included in Storage

File: contracts/erc734/ERC734.sol [hunk 4]
```diff
@@ -1,21 +1,54 @@
...
+    * @dev Definition of the structure of a Key.
+    *
+    * Specification: Keys are cryptographic public keys, or contract addresses associated with this identity.
+    * The structure should be as follows:
+    *   - key: A public key owned by this identity
+    *      - purposes: uint256[] Array of the key purposes, like 1 = MANAGEMENT, 2 = EXECUTION
+    *      - keyType: The type of key used, which would be a uint256 for different key types. e.g. 1 = ECDSA, 2 = RSA, etc.
+    *      - key: bytes32 The public key. // Its the Keccak256 hash of the key
+    */
+    struct Key {
```
Comment (Aboudjem) [inline contracts/erc734/ERC734.sol:20]: both structs should be included in Storage in order to respect the variables and functions order

File: contracts/erc734/ERC734.sol [hunk 5]
```diff
@@ -1,21 +1,54 @@
...
+
+    struct Execution {
+        address to;
+        uint256 value;
+        bytes data;
+        bool approved;
+        bool executed;
+    }
+
+    uint256 public constant MANAGEMENT_KEY = 1;
```
Comment (Aboudjem) [inline contracts/erc734/ERC734.sol:34]: Should we keep these constant variables? Knowing that they are not used (except MANAGEMENT_KEY once)

Comment (Nakasar) [reply contracts/erc734/ERC734.sol:34]: Nope, but let's first take care of this PR that separate the code, then we will handle other updates

**Review by Aboudjem — APPROVED** [review]

**Pull request merged**